rust/libnewsboat/src/configpaths.rs

```diff
@@ -0,0 +1,387 @@
+        };
+
+        let m_env_home = dirs::home_dir();
+        if m_env_home.is_none() {
+            let uid = unsafe { libc::getuid() };
+
+            // TODO: i18n
+            config_paths.m_error_message = format!("Fatal error: couldn't determine home directory!\nPlease set the HOME environment variable or add a valid user for UID {}!", uid);
+
+            return config_paths;
+        }
+
+        // hitting this branch means we found a home directory
+        // we can now safely call unwrap on all other functions in the dirs crate
+
+        config_paths.m_env_home = m_env_home.unwrap();
+        config_paths.find_dirs();
+        config_paths
+    }
+
+    fn migrate_data_from_newsbeuter_xdg(&mut self) -> bool {
+        // This can't panic because we've tested we can find the home directory in ConfigPaths::new
+        // This should be replaced with proper error handling after this is not used by c++ anymore
+        let xdg_config_dir = dirs::config_dir().unwrap();
+        let xdg_data_dir = dirs::data_dir().unwrap();
+
+        let newsbeuter_config_dir = xdg_config_dir.join(NEWSBEUTER_SUBDIR_XDG);
+        let newsbeuter_data_dir = xdg_data_dir.join(NEWSBEUTER_SUBDIR_XDG);
+
+        let newsboat_config_dir = xdg_config_dir.join(NEWSBOAT_SUBDIR_XDG);
+        let newsboat_data_dir = xdg_data_dir.join(NEWSBOAT_SUBDIR_XDG);
+
+        if !newsbeuter_config_dir.is_dir() {
+            return false;
+        }
+
+        fn exists(path: &Path) -> bool {
+            let exists = path.exists();
+            if exists {
+                log!(
+                    Level::Debug,
+                    "{:?} already exists, aborting XDG migration.",
+                    path
+                );
+            }
+            return exists;
+        }
+
+        if exists(&newsboat_config_dir) {
+            return false;
+        }
+
+        if exists(&newsboat_data_dir) {
+            return false;
+        }
+
+        self.m_config_dir = newsboat_config_dir;
+        self.m_data_dir = newsboat_data_dir;
+
+        if !self.m_silent {
+            eprintln!("Migrating configs and data from Newsbeuter's XDG dirs...");
+        }
+
+        if !try_mkdir(&self.m_config_dir) {
+            return false;
+        }
+
+        if !try_mkdir(&self.m_data_dir) {
+            return false;
+        }
+
+        // We ignore the return codes because it's okay if some files are missing.
+
+        // in config
+        let _ = migrate_file(&newsbeuter_config_dir, &self.m_config_dir, "urls");
+        let _ = migrate_file(&newsbeuter_config_dir, &self.m_config_dir, "config");
+
+        // in data
+        let _ = migrate_file(&newsbeuter_data_dir, &self.m_data_dir, "cache.db");
+        let _ = migrate_file(&newsbeuter_data_dir, &self.m_data_dir, "queue");
+        let _ = migrate_file(&newsbeuter_data_dir, &self.m_data_dir, "history.search");
+        let _ = migrate_file(&newsbeuter_data_dir, &self.m_data_dir, "history.cmdline");
```

Comment:
```suggestion
        migrate_file(&newsbeuter_data_dir, &self.m_data_dir, "history.cmdline").ok();
```

I believe this may be a better choice?

Reply:
That's what the code used to say before my change, but I found it not entirely obvious. Is the `ok()` approach an idiom?

Reply:
Not really sure about that, but I recall seeing that as an idiom to `let _ = `, cannot find it on clippy yet. There is a `#[must_use]` but I mostly see `env_logger::init().ok()` IIRC

Reply:
I couldn't find any evidence for or against either, and since `let _` looks more explicit to me, I'll keep it for now. Thanks for taking a look, @pickfire!